source/contracts/libraries/collections/Map.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity 0.4.17;
+
+import 'libraries/DelegationTarget.sol';
+import 'libraries/Ownable.sol';
+import 'libraries/Initializable.sol';
+
+
+contract Map is DelegationTarget ,Ownable, Initializable {
+    mapping(bytes32 => bytes32) private items;
+    uint256 private count;
+
+    function initialize(address _owner) public beforeInitialized returns (bool) {
+        endInitialization();
+        owner = _owner;
+        return true;
+    }
+
+    function add(bytes32 _key, bytes32 _value) public onlyOwner returns (bool) {
+        if (contains(_key)) {
+            return false;
+        }
+        items[_key] = _value;
+        count += 1;
+        return true;
+    }
+
+    function addAsAddress(bytes32 _key, address _value) public onlyOwner returns (bool) {
```

Comment:
I don't believe there is a need to have this function named differently.  It can just be `add(bytes32,address)` and the compiler will pick the right one.  This will make it easier on the callsite as they can just pass whatever they have and it will "do the right thing".

Reply:
That at least didn't seem to work properly in testing. It may just be a problem with pyethereum but given that I opted to make it different. If it is just a problem with pyethereum I can make the test helper method name different but keep the map name the same. Thoughts?

Reply:
Ah, yeah.  Pyethereum may struggle with overloads but I'm reasonably confident that Solidity supports function overloading.  My vote would be to allow for the function overloads and create a helper contract if necessary to address the pyethereum issue (or figure out some other way to deal with the fact that Pyethereum is a dynamic language).

Reply:
Will do then!